arch/arm/soc/st_stm32/stm32f7/flash_registers.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018 Yurii Hamann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _STM32F7X_FLASH_REGISTERS_H_
+#define _STM32F7X_FLASH_REGISTERS_H_
+
+/**
+ * @brief
+ *
+ * Based on reference manual:
+ *   RM0385 Reference manual STM32F75xxx and STM32F74xxx
+ *   advanced ARM(r)-based 32-bit MCUs
+ *
+ * Chapter 3: Embedded Flash Memory
+ */
+
+union __flash_acr {
+	u32_t val;
+	struct {
+		u32_t latency :4 __packed;
+		u32_t rsvd__4_7 :4 __packed;
+		u32_t prften :1 __packed;
+		u32_t icen :1 __packed;
```

Comment:
This seems to have been copied from the STM32F4 structure. On the STM32F7, the caches are included in the Cortex-M7, so the fields for controlling the caches are not there. This should be instead:

```
union __flash_acr {
        u32_t val;
        struct {
                u32_t latency :4 __packed;
                u32_t rsvd__4_7 :4 __packed;
                u32_t prften :1 __packed;
                u32_t arten :1 __packed;
                u32_t rsvd__10 :1 __packed;
                u32_t artrst :1 __packed;
                u32_t rsvd__12_31 :20 __packed;
        } bit;
};

```

Reply:
I will check it.

Reply:
I will check it.

Reply:
Thank you. You are right.